### PR TITLE
Add TCP check option and fix descriptions

### PR DESCRIPTION
**Pull request by juho9000**
##### SUMMARY

Added TCP check option, fixed the module option descriptions to be in line with instructions (beginning with a capital letter)

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

consul module
##### ANSIBLE VERSION

```
ansible 2.4.0
  config file =
  configured module search path = [u'/Users/juho9000/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Oct 11 2016, 05:20:59) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]
```

### CODE REVIEW

File: lib/ansible/modules/clustering/consul.py [hunk 1]
```diff
@@ -151,12 +151,20 @@
...
+          - Checks can be registered with an http endpoint. This means that consul
             will check that the http endpoint returns a successful http status.
             Interval must also be provided with this option.
         required: false
         default: None
         version_added: "2.0"
+    tcp:
+        description:
+          - TCP check tries to make a TCP connection attempt at set intervals
+            to the specified IP or hostname. Interval must also be prived with
```
Comment (sgargan) [inline lib/ansible/modules/clustering/consul.py]: provided?

File: lib/ansible/modules/clustering/consul.py [hunk 2]
```diff
@@ -495,6 +513,13 @@ def __init__(self, check_id, name, node=None, host='localhost',
 
             self.check = consul.Check.http(http, self.interval, self.timeout)
 
+        if tcp:
+            if interval is None:
+                raise Exception('tcp check must specify interval')
+            tcp_host, tcp_port = tcp.split(':')
```
Comment (sgargan) [inline lib/ansible/modules/clustering/consul.py]: this will fail if there is no ':', it should probably check and emit a warning.

File: lib/ansible/modules/clustering/consul.py [hunk 3]
```diff
@@ -192,6 +200,13 @@
     interval: 60s
     http: http://localhost:80/status
 
+- name: register nginx with a tcp check
+  consul:
+    service_name: nginx
```
Comment (sgargan) [inline lib/ansible/modules/clustering/consul.py]: nginx is something you'd most likely check with the http check, might be worth changing this to a non-http service to illustrate the difference.

**Review by sgargan — COMMENTED** [review]
@juho9000 thanks for adding this. Looks good, I've a couple of small comments.

File: lib/ansible/modules/clustering/consul.py [hunk 4]
```diff
@@ -277,10 +279,12 @@ def add(module):
 
 def remove(module):
     ''' removes a service or a check '''
-    service_id = module.params.get('service_id') or module.params.get('service_name')
+    service_id = module.params.get(
+        'service_id') or module.params.get('service_name')
```
Comment (sgargan) [inline lib/ansible/modules/clustering/consul.py]: I'd avoid reformatting the code in a pr like this, formatting is pretty subjective and this is not related to your change. If you're adamant about the formatting please submit a separate pr.

**Comment by juho9000** [comment]
@sgargan removed the formatting-related commit altogether

**Comment by sgargan** [comment]
looks good, nice one. shipit

**Review by vegai — APPROVED** [review]
Looking good

**Comment by calfonso** [comment]
@juho9000 this still needs to be rebased.

**Comment by gundalow** [comment]
Hi,
This PR seems to have gotten stuck, so we are going to close it. We like to include community contributions, so this is a difficult call for us. Here's why we're closing this one:
* It's a feature
* It was created from an old version of the code
* We want to clear out the backlog so it's easier to find bug fixes
* There haven't been any comments or other activity in a long time

If you're the PR author, and you want to start the conversation back up, please:
* reopen the PR and
* rebase the branch

If you're a community member, and you want to see this change merged, please:
* Create a new PR and 
* Reference this author in the git commit & PR and
* Reference this PR in the new PR

How to rebase a branch: https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html
How to communicate with the Ansible community: https://docs.ansible.com/ansible/devel/community/communication.html
How to contribute to Ansible: https://docs.ansible.com/ansible/devel/community/index.html
